src/key.rs

```diff
@@ -106,7 +107,7 @@ pub trait DcKey: Serialize + Deserializable + KeyTrait + Clone {
 
     /// The fingerprint for the key.
     fn fingerprint(&self) -> Fingerprint {
-        Fingerprint::new(KeyTrait::fingerprint(self))
+        Fingerprint::new_unchecked(KeyTrait::fingerprint(self))
```

Comment:
This, and the existence of the constructor altogether, is perhaps an odd choice.  I'm rather undecided on it.